vllm/model_executor/models/utils.py

```diff
@@ -564,6 +564,32 @@ def is_pp_missing_parameter(name: str, model: torch.nn.Module) -> bool:
         for missing_layer_name in get_pp_missing_layer_names(model))
 
 
+def get_inputs_embeds(
+    input_ids: Optional[torch.Tensor],
+    embeddings_module: Callable[[torch.Tensor], torch.Tensor],
+    inputs_embeds: Optional[torch.Tensor] = None,
+    inputs_embeds_masks: Optional[torch.Tensor] = None,
+) -> torch.Tensor:
+    """Get the input embeddings from either `input_ids` and `inputs_embeds`."""
+    if inputs_embeds is not None:
+        if inputs_embeds_masks is None:
+            hidden_states = inputs_embeds
+        else:
+            msg = "inputs_embeds should not be masked out for multimodal models"
+            assert input_ids is not None, msg
+
+            hidden_states = embeddings_module(input_ids)
+            hidden_states[inputs_embeds_masks] = inputs_embeds
```

Comment:
this line is not compatible with compilation, the shape of `inputs_embeds` does not match `hidden_states` .

Reply:
So basically, in-place operations aren't allowed?

Reply:
in-place is fine, but we need all operations to have the same batchsize.